File: packages/extract/test/fixtures/polyglot/swift/Job.swift
import Foundation

enum State { case idle, running, done, failed, retrying }

class Job {
    var state: State = .idle

    func start() {
        guard state == .idle else { return }
        state = .running
    }

    func finish() {
        if state == .running {
            state = .done
        }
    }

    func fail() {
        state = .failed
    }
}

File: packages/extract/test/fixtures/polyglot/php/Job.php
<?php

enum State { case Idle; case Running; case Done; case Failed; case Retrying; }

class Job {
    private State $state = State::Idle;

    public function start(): void {
        if ($this->state !== State::Idle) {
            return;
        }
        $this->state = State::Running;
    }

    public function finish(): void {
        if ($this->state === State::Running) {
            $this->state = State::Done;
        }
    }

    public function fail(): void {
        $this->state = State::Failed;
    }
}
